File: apple/Sources/ReactNativeWatchOS/Modifiers/ProjectionEffect.swift
import SwiftUI

/// SwiftUI `.projectionEffect(_:)`. LIMITED: only the 2D affine subset is
/// bridged. The `ProjectionTransform` is built from a `CGAffineTransform`
/// (same `a,b,c,d,tx,ty` params as `transformEffect`) via
/// `ProjectionTransform(_ m: CGAffineTransform)`. Full 3D `CATransform3D`
/// projections cannot be expressed across the JS bridge and are NOT
/// supported here.
enum RNWProjectionEffectModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("projectionEffect") { view, params, _ in
            let transform = RNWAffineTransform.make(from: params)
            return AnyView(view.projectionEffect(ProjectionTransform(transform)))
        }
    }
}

File: src/modifiers/projectionEffect.ts
import { createModifier } from './createModifier';
import type { AffineTransformParams } from './transformEffect';

/// SwiftUI `.projectionEffect(_:)`. Bridged via the 2D affine subset:
/// a `ProjectionTransform` is built from `CGAffineTransform(a:b:c:d:tx:ty:)`
/// (same six components as `transformEffect`). Full 3D `CATransform3D`
/// projections are NOT bridged — only affine projections are expressible
/// here. Omitted components default to the identity matrix.
export function projectionEffect(params: AffineTransformParams) {
  return createModifier('projectionEffect', params);
}

export type { AffineTransformParams };
